### ISSUE TITLE
Improve moderation notifications

### ISSUE THREAD

**Issue by cadfile** [postmatic/epoch#38]
I have my blog set to "Comment author must have a previously approved comment" so that new people have their comments moderated. Epoch ignores that setting and publishes the comment on the site. It still shows up in my moderation queue but hasn't been approved

**Comment by Vernal** [comment]
Hey Cadfile,

I don't understand what you mean. Could you rephrase the issue? Your statements contradict:
1. Epoch ignores that setting and publishes the comment on the site.
1. It still shows up in my moderation queue but hasn't been approved.

Those two statements contradict. Do you mean to say that Epoch is showing unmoderated comments on the frontend?

Best,
Jason

**Comment by cadfile** [comment]
Sorry for the confusion.

Yes Epoch is showing unmoderated comments on the frontend. I don't moderate all comments just from people who have not had a comment approved before.

**Comment by Vernal** [comment]
The plugin may still show a comment which is in moderation if it was entered by you. 

Can you confirm that the comments which are slipping through were left by other users other than your own?

To rephrase: if you enter a comment and it is being held for moderation you may still see it on the front end, in the context of the conversation, while it is pending.

**Comment by cadfile** [comment]
I always test these things using a separate browser where I am not logged in and I use fake name info

Here is screenshot showing my dashboard telling me the comment needs to be moderated yet it is showing on the post

![unmoderatedposted](https://cloud.githubusercontent.com/assets/4371578/9024046/e54d2954-3885-11e5-8504-021fa01ad90d.png)

**Comment by cadfile** [comment]
I figured it out. The ajax isn't refreshing the page so when the page is refreshed the comment doesn't show on the page before it is approved. When I approve it and refresh the post page then the comment shows up.

Might cause confusion for users who think their comment was posted then comes back and it isn't there since it didn't get approved yet. Might need a message saying comment submitted for moderation or something like that.

**Comment by Vernal** [comment]
Thanks, cadfile. It looks like things are working as designed but we need to improve moderation notifications. Noted!

Jason

**Comment by mrjarbenne** [comment]
P2 does a nice job with this. The comments show up using ajax (like epoch) but the comment is tagged with a line letting the author know that the comment is awaiting moderation: 

![2015-08-07_15-35-58](https://cloud.githubusercontent.com/assets/480667/9144333/24bead68-3d1a-11e5-95de-c5242632a7a9.png)
